// contentful-cli/dist/cmds/config_cmds/remove.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.handler = exports.removeHandler = exports.builder = exports.desc = exports.command = undefined;

var _async = require('../../utils/async');

var _log = require('../../utils/log');

var _emojis = require('../../utils/emojis.js');

var _context = require('../../context');

var _lodash = require('lodash');

const command = exports.command = 'remove';

const desc = exports.desc = 'Removes a config from ~/.contentfulrc.json';

const builder = exports.builder = yargs => {
  return yargs.usage('Usage: contentful config remove [options]').option('management-token', {
    alias: 'mt',
    describe: 'Remove the API management token from the config',
    type: 'boolean',
    default: false
  }).option('active-space-id', {
    alias: 'as',
    describe: 'Remove the active space id from the config',
    type: 'boolean',
    default: false
  }).option('active-environment-id', {
    alias: 'ae',
    describe: 'Remove the active environment id from the config'
  }).option('host', {
    alias: 'ho',
    describe: 'Remove the management host from the config'
  }).option('proxy', {
    alias: 'p',
    describe: 'Remove the proxy from the config',
    type: 'boolean',
    default: false
  }).option('raw-proxy', {
    alias: 'rp',
    describe: 'Pass proxy config as raw config instead of creating a httpsAgent',
    type: 'boolean'
  }).option('all', {
    describe: 'Remove all the things from the config',
    type: 'boolean',
    default: false
  });
};

const removeHandler = exports.removeHandler = async argv => {
  let context = await (0, _context.getContext)();
  const optionsToPick = ['cmaToken', 'proxy', 'activeSpaceId'];
  let options = (0, _lodash.pick)(context, optionsToPick);
  if (argv.all) {
    options = {};
  } else {
    if (argv.managementToken) {
      delete options['cmaToken'];
    }
    if (argv.activeSpaceId) {
      delete options['activeSpaceId'];
    }
    if (argv.proxy) {
      delete options['proxy'];
    }
  }
  (0, _context.emptyContext)();
  (0, _context.setContext)(options);
  await (0, _context.storeRuntimeConfig)();
  (0, _log.success)(`${_emojis.successEmoji} config removed successfully`);
};

const handler = exports.handler = (0, _async.handleAsyncError)(removeHandler);
