// @contentful/forma-36-react-components/dist/components/Card/ReferenceCard/ReferenceCard.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ReferenceCard = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _truncate = _interopRequireDefault(require("truncate"));

var _Card = _interopRequireDefault(require("../Card"));

var _Tag = _interopRequireDefault(require("../../Tag"));

var _ReferenceCardSkeleton = _interopRequireDefault(require("./ReferenceCardSkeleton"));

var styles = {
  "ReferenceCard": "ReferenceCard__ReferenceCard___316Zf",
  "ReferenceCard--is-loading": "ReferenceCard__ReferenceCard--is-loading___3iq8p",
  "ReferenceCard__meta": "ReferenceCard__ReferenceCard__meta___1m-5R",
  "ReferenceCard__content-type": "ReferenceCard__ReferenceCard__content-type___2Yvxm",
  "ReferenceCard__actions": "ReferenceCard__ReferenceCard__actions___heiRn",
  "ReferenceCard__content": "ReferenceCard__ReferenceCard__content___2hzPx",
  "ReferenceCard__body": "ReferenceCard__ReferenceCard__body___1QMbM",
  "ReferenceCard__title": "ReferenceCard__ReferenceCard__title___23s0u",
  "ReferenceCard__description": "ReferenceCard__ReferenceCard__description___3i-fh",
  "ReferenceCard__thumbnail": "ReferenceCard__ReferenceCard__thumbnail___3gmgs"
};

var ReferenceCard =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(ReferenceCard, _Component);

  function ReferenceCard() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ReferenceCard);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ReferenceCard)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderTitle", function (title) {
      var truncatedTitle = (0, _truncate.default)(title, 255, {});
      return _react.default.createElement("h1", {
        title: title.length > 255 ? title : '',
        className: styles.ReferenceCard__title,
        "data-test-id": "title"
      }, truncatedTitle);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderDescription", function (description) {
      var truncatedDescription = (0, _truncate.default)(description, 95, {});
      return _react.default.createElement("p", {
        className: styles.ReferenceCard__description
      }, truncatedDescription);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderThumbnail", function (thumbnailElement) {
      return _react.default.createElement("figure", {
        className: styles.ReferenceCard__thumbnail
      }, thumbnailElement);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderActionElements", function (actionElements) {
      return _react.default.createElement("div", {
        className: styles.ReferenceCard__actions
      }, actionElements);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderStatus", function (status) {
      var label;
      var type;

      switch (status) {
        case 'archived':
          label = 'archived';
          type = 'negative';
          break;

        case 'changed':
          label = 'changed';
          type = 'primary';
          break;

        case 'published':
          label = 'published';
          type = 'positive';
          break;

        default:
          label = 'draft';
          type = 'warning';
      }

      return _react.default.createElement(_Tag.default, {
        tagType: type
      }, label);
    });
    return _this;
  }

  (0, _createClass2.default)(ReferenceCard, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          extraClassNames = _this$props.extraClassNames,
          title = _this$props.title,
          onClick = _this$props.onClick,
          testId = _this$props.testId,
          description = _this$props.description,
          contentType = _this$props.contentType,
          status = _this$props.status,
          thumbnailElement = _this$props.thumbnailElement,
          loading = _this$props.loading,
          actionElements = _this$props.actionElements,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["extraClassNames", "title", "onClick", "testId", "description", "contentType", "status", "thumbnailElement", "loading", "actionElements"]);
      var classNames = (0, _classnames.default)(styles.ReferenceCard, extraClassNames);
      return _react.default.createElement(_Card.default, (0, _extends2.default)({
        extraClassNames: classNames,
        onClick: !loading ? onClick : undefined,
        testId: testId
      }, otherProps), loading ? _react.default.createElement(_ReferenceCardSkeleton.default, null) : _react.default.createElement("article", {
        className: styles.ReferenceCard__wrapper
      }, _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("div", {
        className: styles.ReferenceCard__meta
      }, _react.default.createElement("div", {
        className: styles['ReferenceCard__content-type'],
        "data-test-id": "content-type"
      }, contentType), status && this.renderStatus(status), actionElements && this.renderActionElements(actionElements)), _react.default.createElement("div", {
        className: styles.ReferenceCard__content
      }, _react.default.createElement("div", {
        className: styles.ReferenceCard__body
      }, title && this.renderTitle(title), description && this.renderDescription(description)), thumbnailElement && this.renderThumbnail(thumbnailElement)))));
    }
  }]);
  return ReferenceCard;
}(_react.Component);

exports.ReferenceCard = ReferenceCard;
ReferenceCard.displayName = "ReferenceCard";
(0, _defineProperty2.default)(ReferenceCard, "propTypes", {
  extraClassNames: _propTypes.default.string,
  title: _propTypes.default.string,
  testId: _propTypes.default.string,
  description: _propTypes.default.string,
  contentType: _propTypes.default.string,
  status: _propTypes.default.oneOf(['archived', 'changed', 'draft', 'published']),
  thumbnailElement: _propTypes.default.node,
  loading: _propTypes.default.bool,
  onClick: _propTypes.default.func,
  actionElements: _propTypes.default.node
});
(0, _defineProperty2.default)(ReferenceCard, "defaultProps", {
  title: 'Untitled',
  description: undefined,
  testId: 'cf-ui-reference-card',
  contentType: undefined,
  status: undefined,
  thumbnailElement: undefined,
  loading: undefined,
  onClick: undefined,
  actionElements: undefined,
  extraClassNames: undefined
});
var _default = ReferenceCard;
exports.default = _default;
