// @contentful/forma-36-react-components/dist/components/Card/InlineReferenceCard/InlineReferenceCard.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.InlineReferenceCard = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactTransitionGroup = require("react-transition-group");

var _IconButton = _interopRequireDefault(require("../../IconButton"));

var _Dropdown = _interopRequireDefault(require("../../Dropdown/Dropdown"));

var _DropdownList = _interopRequireDefault(require("../../Dropdown/DropdownList"));

var _Card = _interopRequireDefault(require("../Card"));

var _InlineReferenceCardSkeleton = _interopRequireDefault(require("./InlineReferenceCardSkeleton"));

var styles = {
  "InlineReferenceCard": "InlineReferenceCard__InlineReferenceCard___1FGH9",
  "InlineReferenceCard__skeleton-wrapper": "InlineReferenceCard__InlineReferenceCard__skeleton-wrapper___nBSht",
  "InlineReferenceCard__text-wrapper": "InlineReferenceCard__InlineReferenceCard__text-wrapper___2GOVz",
  "InlineReferenceCard__status-indicator": "InlineReferenceCard__InlineReferenceCard__status-indicator___1pYa8",
  "InlineReferenceCard__status-indicator--published": "InlineReferenceCard__InlineReferenceCard__status-indicator--published___38UbM",
  "InlineReferenceCard__status-indicator--draft": "InlineReferenceCard__InlineReferenceCard__status-indicator--draft___2ziJv",
  "InlineReferenceCard__status-indicator--archived": "InlineReferenceCard__InlineReferenceCard__status-indicator--archived___1YxwY",
  "InlineReferenceCard__status-indicator--changed": "InlineReferenceCard__InlineReferenceCard__status-indicator--changed___U_rmd",
  "InlineReferenceCard__dropdown": "InlineReferenceCard__InlineReferenceCard__dropdown___3YX72",
  "InlineReferenceCard__spinner--enter": "InlineReferenceCard__InlineReferenceCard__spinner--enter___1M-BG",
  "InlineReferenceCard__spinner--enter-active": "InlineReferenceCard__InlineReferenceCard__spinner--enter-active___27jZB",
  "InlineReferenceCard__spinner--exit": "InlineReferenceCard__InlineReferenceCard__spinner--exit___2Bwsk",
  "InlineReferenceCard__spinner--exit-active": "InlineReferenceCard__InlineReferenceCard__spinner--exit-active___1JuBb"
};

var InlineReferenceCard =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(InlineReferenceCard, _Component);

  function InlineReferenceCard() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, InlineReferenceCard);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(InlineReferenceCard)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      isDropdownOpen: false
    });
    return _this;
  }

  (0, _createClass2.default)(InlineReferenceCard, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          extraClassNames = _this$props.extraClassNames,
          dropdownListItemNodes = _this$props.dropdownListItemNodes,
          isSelected = _this$props.isSelected,
          children = _this$props.children,
          testId = _this$props.testId,
          isLoading = _this$props.isLoading,
          status = _this$props.status,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["extraClassNames", "dropdownListItemNodes", "isSelected", "children", "testId", "isLoading", "status"]);
      var classNames = (0, _classnames.default)(styles.InlineReferenceCard, extraClassNames);
      var statusIndicatorClassNames = (0, _classnames.default)(styles['InlineReferenceCard__status-indicator'], (0, _defineProperty2.default)({}, styles["InlineReferenceCard__status-indicator--".concat(status)], status && !isLoading));
      return _react.default.createElement(_Card.default, (0, _extends2.default)({
        selected: isSelected,
        extraClassNames: classNames
      }, otherProps, {
        "data-test-id": testId
      }), _react.default.createElement(_reactTransitionGroup.CSSTransition, {
        timeout: 100,
        in: isLoading,
        classNames: {
          enter: styles['InlineReferenceCard__spinner--enter'],
          enterActive: styles['InlineReferenceCard__spinner--enter-active'],
          exit: styles['InlineReferenceCard__spinner--exit'],
          exitActive: styles['InlineReferenceCard__spinner--exit-active']
        },
        mountOnEnter: true,
        unmountOnExit: true
      }, _react.default.createElement("div", {
        className: styles['InlineReferenceCard__skeleton-wrapper']
      }, _react.default.createElement(_InlineReferenceCardSkeleton.default, null))), _react.default.createElement("div", {
        className: statusIndicatorClassNames
      }), _react.default.createElement("span", {
        className: styles['InlineReferenceCard__text-wrapper']
      }, isLoading ? 'Loading' : children), dropdownListItemNodes && _react.default.createElement(_Dropdown.default, {
        onClose: function onClose() {
          _this2.setState({
            isDropdownOpen: false
          });
        },
        position: "bottom-right",
        extraClassNames: styles.InlineReferenceCard__dropdown,
        isOpen: this.state.isDropdownOpen,
        toggleElement: _react.default.createElement(_IconButton.default, {
          extraClassNames: styles['InlineReferenceCard__icon-button'],
          iconProps: {
            icon: 'MoreHorizontal'
          },
          buttonType: "secondary",
          label: "Inline reference actions",
          onClick: function onClick() {
            _this2.setState(function (state) {
              return {
                isDropdownOpen: !state.isDropdownOpen
              };
            });
          }
        })
      }, _react.default.createElement(_DropdownList.default, {
        extraClassNames: styles['InlineReferenceCard__dropdown-list']
      }, dropdownListItemNodes)));
    }
  }]);
  return InlineReferenceCard;
}(_react.Component);

exports.InlineReferenceCard = InlineReferenceCard;
InlineReferenceCard.displayName = "InlineReferenceCard";
(0, _defineProperty2.default)(InlineReferenceCard, "propTypes", {
  extraClassNames: _propTypes.default.string,
  children: _propTypes.default.node.isRequired,
  isSelected: _propTypes.default.bool,
  dropdownListItemNodes: _propTypes.default.node,
  isLoading: _propTypes.default.bool,
  testId: _propTypes.default.string,
  status: _propTypes.default.oneOf(['archived', 'changed', 'draft', 'published'])
});
var _default = InlineReferenceCard;
exports.default = _default;
