// @contentful/forma-36-react-components/dist/components/Table/Table/Table.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Table = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "Table": "Table__Table___2x3zV"
};

var Table =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Table, _Component);

  function Table() {
    (0, _classCallCheck2.default)(this, Table);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Table).apply(this, arguments));
  }

  (0, _createClass2.default)(Table, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          extraClassNames = _this$props.extraClassNames,
          children = _this$props.children,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["extraClassNames", "children"]);
      return _react.default.createElement("table", (0, _extends2.default)({
        className: (0, _classnames.default)(extraClassNames, styles['Table']),
        cellPadding: "0",
        cellSpacing: "0"
      }, otherProps), children);
    }
  }]);
  return Table;
}(_react.Component);

exports.Table = Table;
Table.displayName = "Table";
(0, _defineProperty2.default)(Table, "propTypes", {
  extraClassNames: _propTypes.default.string,
  style: _propTypes.default.any,
  children: _propTypes.default.node.isRequired
});
var _default = Table;
exports.default = _default;
