// contentful-migration/built/lib/intent/content-type-delete.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const base_intent_1 = require("./base-intent");
const content_type_unpublish_1 = require("../action/content-type-unpublish");
const content_type_delete_1 = require("../action/content-type-delete");
const chalk_1 = require("chalk");
class ContentTypeDeleteIntent extends base_intent_1.default {
    isContentTypeDelete() {
        return true;
    }
    groupsWith() {
        return false;
    }
    endsGroup() {
        return true;
    }
    shouldSave() {
        return false;
    }
    shouldPublish() {
        return false;
    }
    toActions() {
        return [
            new content_type_unpublish_1.ContentTypeUnpublishAction(this.getContentTypeId()),
            new content_type_delete_1.ContentTypeDeleteAction(this.getContentTypeId())
        ];
    }
    toPlanMessage() {
        return {
            heading: chalk_1.default `Delete Content Type {bold.yellow ${this.getContentTypeId()}}`,
            details: [],
            sections: []
        };
    }
}
exports.default = ContentTypeDeleteIntent;
