// contentful-migration/built/lib/intent/entry-transform-to-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const base_intent_1 = require("./base-intent");
const entry_transform_to_type_1 = require("../action/entry-transform-to-type");
const chalk_1 = require("chalk");
class EntryTransformToTypeIntent extends base_intent_1.default {
    getRelatedContentTypeIds() {
        return [
            this.payload.entryTransformationToType.sourceContentType
        ];
    }
    isEntryTransformToType() {
        return true;
    }
    endsGroup() {
        return true;
    }
    groupsWith() {
        return false;
    }
    toActions() {
        return [
            new entry_transform_to_type_1.EntryTransformToTypeAction(this.payload.entryTransformationToType)
        ];
    }
    toPlanMessage() {
        return {
            heading: chalk_1.default `Migrate entries from {bold.yellow ${this.getContentTypeId()}}`,
            details: [
                `from: ${this.payload.entryTransformationToType.sourceContentType}`,
                `to: ${this.payload.entryTransformationToType.targetContentType}`
            ],
            sections: []
        };
    }
    shouldSave() {
        return false;
    }
    shouldPublish() {
        return false;
    }
    requiresAllEntries() {
        return this.payload.entryTransformationToType.updateReferences;
    }
}
exports.default = EntryTransformToTypeIntent;
