// @contentful/forma-36-react-components/dist/components/RadioButton/RadioButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RadioButton = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _ControlledInput = _interopRequireDefault(require("../ControlledInput/ControlledInput"));

var RadioButton = function RadioButton(props) {
  return _react.default.createElement(_ControlledInput.default, (0, _extends2.default)({}, props, {
    type: "radio"
  }));
};

exports.RadioButton = RadioButton;
RadioButton.displayName = "RadioButton";
RadioButton.defaultProps = {
  id: undefined,
  extraClassNames: undefined,
  onChange: undefined,
  onBlur: undefined,
  onFocus: undefined,
  checked: undefined,
  value: undefined,
  name: undefined,
  required: false,
  disabled: false,
  type: 'checkbox',
  testId: 'cf-ui-radio-button'
};
var _default = RadioButton;
exports.default = _default;
