// @contentful/forma-36-react-components/dist/components/Card/AssetCard/AssetCardSkeleton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SkeletonContainer = _interopRequireDefault(require("../../Skeleton/SkeletonContainer"));

var _SkeletonBodyText = _interopRequireDefault(require("../../Skeleton/SkeletonBodyText"));

var _SkeletonImage = _interopRequireDefault(require("../../Skeleton/SkeletonImage"));

var AssetCardSkeleton = function AssetCardSkeleton() {
  return _react.default.createElement(_SkeletonContainer.default, {
    svgWidth: 240,
    clipId: "f36-asset-card-skeleton"
  }, _react.default.createElement(_SkeletonImage.default, {
    offsetLeft: 85,
    offsetTop: 100
  }), _react.default.createElement(_SkeletonBodyText.default, {
    offsetLeft: 70,
    offsetTop: 190,
    numberOfLines: 1,
    width: 100
  }));
};

AssetCardSkeleton.displayName = "AssetCardSkeleton";
var _default = AssetCardSkeleton;
exports.default = _default;
