// @contentful/forma-36-react-components/dist/components/Form/FieldGroup/FieldGroup.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.FieldGroup = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "FieldGroup": "FieldGroup__FieldGroup___2mLmO",
  "FieldGroup__item": "FieldGroup__FieldGroup__item___2qkC3",
  "FieldGroup--row": "FieldGroup__FieldGroup--row___17yyV"
};

var FieldGroup =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(FieldGroup, _Component);

  function FieldGroup() {
    (0, _classCallCheck2.default)(this, FieldGroup);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(FieldGroup).apply(this, arguments));
  }

  (0, _createClass2.default)(FieldGroup, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          extraClassNames = _this$props.extraClassNames,
          children = _this$props.children,
          row = _this$props.row,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["extraClassNames", "children", "row"]);
      var classNames = (0, _classnames.default)(styles.FieldGroup, styles.extraClassNames, (0, _defineProperty2.default)({}, styles['FieldGroup--row'], row));
      return _react.default.createElement("div", (0, _extends2.default)({
        className: classNames
      }, otherProps), _react.default.Children.map(children, function (child) {
        return _react.default.createElement("div", {
          className: styles.FieldGroup__item
        }, child);
      }));
    }
  }]);
  return FieldGroup;
}(_react.Component);

exports.FieldGroup = FieldGroup;
FieldGroup.displayName = "FieldGroup";
(0, _defineProperty2.default)(FieldGroup, "propTypes", {
  extraClassNames: _propTypes.default.string,
  children: _propTypes.default.node.isRequired,
  row: _propTypes.default.bool
});
(0, _defineProperty2.default)(FieldGroup, "defaultProps", {
  row: false
});
var _default = FieldGroup;
exports.default = _default;
