// @contentful/forma-36-react-components/dist/components/CopyButton/CopyButton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.CopyButton = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _reactCopyToClipboard = _interopRequireDefault(require("react-copy-to-clipboard"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = _interopRequireDefault(require("../Icon"));

var _Tooltip = _interopRequireDefault(require("../Tooltip"));

var _TabFocusTrap = _interopRequireDefault(require("../TabFocusTrap"));

var styles = {
  "CopyButton": "CopyButton__CopyButton___OoA5D",
  "CopyButton__button": "CopyButton__CopyButton__button___52Bc0 a11y__focus-outline--default___2hwb1",
  "CopyButton__text": "CopyButton__CopyButton__text___269PQ helpers__sr-only___3Kv3z",
  "CopyButton__TabFocusTrap": "CopyButton__CopyButton__TabFocusTrap___1Q_DQ"
};

var CopyButton =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(CopyButton, _Component);

  function CopyButton() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, CopyButton);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(CopyButton)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      copied: false
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "copyButton", null);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "tooltipAnchor", null);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onCopy", function (e) {
      _this.props.onCopy(e);

      _this.setState({
        copied: true
      });

      setTimeout(function () {
        _this.setState({
          copied: false
        });

        _this.copyButton.blur();
      }, 1000);
    });
    return _this;
  }

  (0, _createClass2.default)(CopyButton, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          copyValue = _this$props.copyValue,
          extraClassNames = _this$props.extraClassNames,
          testId = _this$props.testId,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["copyValue", "extraClassNames", "testId"]);
      var classNames = (0, _classnames.default)(styles['CopyButton'], extraClassNames);
      return _react.default.createElement("div", (0, _extends2.default)({
        ref: function ref(_ref2) {
          _this2.tooltipAnchor = _ref2;
        },
        className: classNames,
        id: "copyButton",
        "data-test-id": testId
      }, otherProps), _react.default.createElement(_reactCopyToClipboard.default, {
        text: copyValue,
        onCopy: this.onCopy
      }, _react.default.createElement(_Tooltip.default, {
        content: this.state.copied ? 'Copied!' : _react.default.createElement("span", null, "Copy to ", _react.default.createElement("br", null), " clipboard")
      }, _react.default.createElement("button", {
        type: "button",
        ref: function ref(_ref) {
          _this2.copyButton = _ref;
        },
        className: styles['CopyButton__button']
      }, _react.default.createElement(_TabFocusTrap.default, {
        extraClassNames: styles['CopyButton__TabFocusTrap']
      }, _react.default.createElement("span", {
        className: styles['CopyButton__text']
      }, "Copy ", copyValue, " to clipboard"), _react.default.createElement(_Icon.default, {
        icon: "Copy",
        color: "muted"
      }))))));
    }
  }]);
  return CopyButton;
}(_react.Component);

exports.CopyButton = CopyButton;
CopyButton.displayName = "CopyButton";
(0, _defineProperty2.default)(CopyButton, "propTypes", {
  extraClassNames: _propTypes.default.string,
  testId: _propTypes.default.string,
  copyValue: _propTypes.default.string,
  onCopy: _propTypes.default.func
});
(0, _defineProperty2.default)(CopyButton, "defaultProps", {
  testId: 'cf-ui-copy-button',
  onCopy: function onCopy() {}
});
var _default = CopyButton;
exports.default = _default;
