// @contentful/forma-36-react-components/dist/components/Table/TableHead/TableHead.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TableHead = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TableCell = require("../TableCell");

var styles = {
  "TableHead--sticky": "TableHead__TableHead--sticky___1YzE0"
};

var TableHead =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(TableHead, _Component);

  function TableHead() {
    (0, _classCallCheck2.default)(this, TableHead);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TableHead).apply(this, arguments));
  }

  (0, _createClass2.default)(TableHead, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          extraClassNames = _this$props.extraClassNames,
          offsetTop = _this$props.offsetTop,
          isSticky = _this$props.isSticky,
          children = _this$props.children;
      var classNames = (0, _classnames.default)(extraClassNames, (0, _defineProperty2.default)({}, styles["TableHead--sticky"], isSticky));
      return _react.default.createElement(_TableCell.TableCellContext.Provider, {
        value: (0, _objectSpread2.default)({}, _TableCell.contextOptions.head, {
          offsetTop: offsetTop
        })
      }, _react.default.createElement("thead", {
        className: classNames
      }, children));
    }
  }]);
  return TableHead;
}(_react.Component);

exports.TableHead = TableHead;
TableHead.displayName = "TableHead";
(0, _defineProperty2.default)(TableHead, "propTypes", {
  extraClassNames: _propTypes.default.string,
  isSticky: _propTypes.default.bool,
  offsetTop: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
  children: _propTypes.default.node.isRequired
});
(0, _defineProperty2.default)(TableHead, "defaultProps", {
  isSticky: false
});
var _default = TableHead;
exports.default = _default;
