// contentful-cli/dist/cmds/extension_cmds/list.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.handler = exports.builder = exports.desc = exports.command = undefined;
exports.listExtensions = listExtensions;

var _cliTable = require('cli-table3');

var _cliTable2 = _interopRequireDefault(_cliTable);

var _context = require('../../context');

var _contentfulClients = require('../../utils/contentful-clients');

var _assertions = require('../../utils/assertions');

var _async = require('../../utils/async');

var _log = require('../../utils/log');

var _pagination = require('../../utils/pagination');

var _pagination2 = _interopRequireDefault(_pagination);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const command = exports.command = 'list';

const desc = exports.desc = 'List all extensions';

const builder = exports.builder = yargs => {
  return yargs.option('management-token', { alias: 'mt', type: 'string', describe: 'Contentful management API token' }).option('space-id', { type: 'string', describe: 'Space id' }).option('environment-id', { type: 'string', describe: 'Environment id', default: 'master' }).epilog('Copyright 2018 Contentful, this is a BETA release');
};

async function listExtensions(argv) {
  await (0, _assertions.assertLoggedIn)(argv);
  await (0, _assertions.assertSpaceIdProvided)(argv);

  const { cmaToken, activeSpaceId } = await (0, _context.getContext)();
  const spaceId = argv.spaceId || activeSpaceId;
  const environmentId = argv.environmentId;

  const client = await (0, _contentfulClients.createManagementClient)({
    accessToken: argv.managementToken || cmaToken,
    feature: 'extension-list'
  });

  const space = await client.getSpace(spaceId);
  const environment = await space.getEnvironment(environmentId);

  const result = await (0, _pagination2.default)({
    client: environment,
    method: 'getUiExtensions',
    query: {
      order: 'extension.name,sys.id'
    }
  });

  if (result.items.length) {
    const extensions = result.items.sort((a, b) => a.extension.name.localeCompare(b.extension.name));

    const table = new _cliTable2.default({
      head: ['Extension Name', 'Extension ID', 'Version']
    });

    extensions.forEach(extension => {
      table.push([extension.extension.name, extension.sys.id, extension.sys.version]);
    });

    (0, _log.log)(table.toString());
    return;
  }

  (0, _log.log)('No extensions found');
}

const handler = exports.handler = (0, _async.handleAsyncError)(listExtensions);
