// @contentful/forma-36-react-components/dist/components/Dropdown/DropdownListItem/DropdownListItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DropdownListItem = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TabFocusTrap = _interopRequireDefault(require("../../TabFocusTrap/TabFocusTrap"));

var styles = {
  "DropdownListItem": "DropdownListItem__DropdownListItem___LOUnP",
  "DropdownListItem__toggle-button": "DropdownListItem__DropdownListItem__toggle-button___cQ3OZ a11y__focus-outline--default___2hwb1",
  "DropdownListItem__submenu-toggle": "DropdownListItem__DropdownListItem__submenu-toggle___1SVw1",
  "DropdownListItem__toggle-button__inner-wrapper": "DropdownListItem__DropdownListItem__toggle-button__inner-wrapper___1kw4m",
  "DropdownListItem--disabled": "DropdownListItem__DropdownListItem--disabled___1txXv",
  "DropdownListItem--active": "DropdownListItem__DropdownListItem--active___21eet",
  "DropdownListItem--title": "DropdownListItem__DropdownListItem--title___CyVKB"
};

var DropdownListItem =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(DropdownListItem, _Component);

  function DropdownListItem() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, DropdownListItem);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(DropdownListItem)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderSubmenuToggle", function () {
      var _this$props = _this.props,
          onClick = _this$props.onClick,
          onEnter = _this$props.onEnter,
          onLeave = _this$props.onLeave,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["onClick", "onEnter", "onLeave"]);
      return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("button", (0, _extends2.default)({
        type: "button",
        "data-test-id": "cf-ui-dropdown-submenu-toggle",
        className: styles['DropdownListItem__toggle-button'],
        onClick: _this.props.onClick,
        onMouseEnter: _this.props.onEnter,
        onFocus: _this.props.onEnter,
        onMouseLeave: _this.props.onLeave
      }, otherProps), _react.default.createElement(_TabFocusTrap.default, {
        extraClassNames: styles['DropdownListItem__toggle-button__inner-wrapper']
      }, _this.props.submenuToggleLabel)), _this.props.children);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderListItem", function () {
      return _this.props.onClick || _this.props.onMouseDown ? _react.default.createElement("button", {
        type: "button",
        "data-test-id": "cf-ui-dropdown-list-item-button",
        className: styles['DropdownListItem__toggle-button'],
        onClick: function onClick(e) {
          if (!_this.props.isDisabled && _this.props.onClick) {
            _this.props.onClick(e);
          }
        },
        onMouseDown: function onMouseDown(e) {
          if (!_this.props.isDisabled && _this.props.onMouseDown) {
            _this.props.onMouseDown(e);
          }
        }
      }, _react.default.createElement(_TabFocusTrap.default, {
        extraClassNames: styles['DropdownListItem__toggle-button__inner-wrapper']
      }, _this.props.children)) : _this.props.children;
    });
    return _this;
  }

  (0, _createClass2.default)(DropdownListItem, [{
    key: "render",
    value: function render() {
      var _cn;

      var _this$props2 = this.props,
          extraClassNames = _this$props2.extraClassNames,
          isDisabled = _this$props2.isDisabled,
          testId = _this$props2.testId,
          isActive = _this$props2.isActive,
          onClick = _this$props2.onClick,
          onMouseDown = _this$props2.onMouseDown,
          submenuToggleLabel = _this$props2.submenuToggleLabel,
          isTitle = _this$props2.isTitle;
      var classNames = (0, _classnames.default)(styles['DropdownListItem'], extraClassNames, (_cn = {}, (0, _defineProperty2.default)(_cn, styles['DropdownListItem__submenu-toggle'], submenuToggleLabel || onClick || onMouseDown), (0, _defineProperty2.default)(_cn, styles['DropdownListItem--disabled'], isDisabled), (0, _defineProperty2.default)(_cn, styles['DropdownListItem--active'], isActive), (0, _defineProperty2.default)(_cn, styles['DropdownListItem--title'], isTitle), _cn));
      return _react.default.createElement("li", {
        className: classNames,
        "data-test-id": testId
      }, submenuToggleLabel ? this.renderSubmenuToggle() : this.renderListItem());
    }
  }]);
  return DropdownListItem;
}(_react.Component);

exports.DropdownListItem = DropdownListItem;
DropdownListItem.displayName = "DropdownListItem";
(0, _defineProperty2.default)(DropdownListItem, "propTypes", {
  extraClassNames: _propTypes.default.string,
  children: _propTypes.default.node.isRequired,
  onClick: _propTypes.default.func,
  testId: _propTypes.default.string,
  onMouseDown: _propTypes.default.func,
  submenuToggleLabel: _propTypes.default.string,
  onFocus: _propTypes.default.func,
  onLeave: _propTypes.default.func,
  onEnter: _propTypes.default.func,
  isDisabled: _propTypes.default.bool,
  isActive: _propTypes.default.bool,
  isTitle: _propTypes.default.bool
});
(0, _defineProperty2.default)(DropdownListItem, "defaultProps", {
  extraClassNames: undefined,
  submenuToggleLabel: undefined,
  onClick: undefined,
  testId: 'cf-ui-dropdown-list-item',
  onMouseDown: undefined,
  onFocus: undefined,
  onLeave: function onLeave() {},
  onEnter: function onEnter() {},
  isDisabled: false,
  isActive: false,
  isTitle: false
});
var _default = DropdownListItem;
exports.default = _default;
