// @contentful/forma-36-react-components/dist/components/ControlledInput/ControlledInput.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ControlledInput = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "ControlledInput": "ControlledInput__ControlledInput___2XK3j",
  "ControlledInput--disabled": "ControlledInput__ControlledInput--disabled___3prPF"
};

var ControlledInput =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(ControlledInput, _Component);

  function ControlledInput() {
    (0, _classCallCheck2.default)(this, ControlledInput);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(ControlledInput).apply(this, arguments));
  }

  (0, _createClass2.default)(ControlledInput, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          extraClassNames = _this$props.extraClassNames,
          id = _this$props.id,
          testId = _this$props.testId,
          required = _this$props.required,
          disabled = _this$props.disabled,
          onFocus = _this$props.onFocus,
          onBlur = _this$props.onBlur,
          name = _this$props.name,
          onChange = _this$props.onChange,
          checked = _this$props.checked,
          value = _this$props.value,
          type = _this$props.type,
          labelText = _this$props.labelText,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["extraClassNames", "id", "testId", "required", "disabled", "onFocus", "onBlur", "name", "onChange", "checked", "value", "type", "labelText"]);
      var classNames = (0, _classnames.default)(styles['ControlledInput'], extraClassNames, (0, _defineProperty2.default)({}, styles['ControlledInput--disabled'], disabled));
      return _react.default.createElement("input", (0, _extends2.default)({
        className: classNames,
        value: value,
        name: name,
        checked: this.props.checked,
        type: type,
        "data-test-id": testId,
        onChange: function (_onChange) {
          function onChange(_x) {
            return _onChange.apply(this, arguments);
          }

          onChange.toString = function () {
            return _onChange.toString();
          };

          return onChange;
        }(function (e) {
          if (onChange) {
            onChange(e);
          }
        }),
        onBlur: function (_onBlur) {
          function onBlur(_x2) {
            return _onBlur.apply(this, arguments);
          }

          onBlur.toString = function () {
            return _onBlur.toString();
          };

          return onBlur;
        }(function (e) {
          if (onBlur) {
            onBlur(e);
          }
        }),
        onFocus: function (_onFocus) {
          function onFocus(_x3) {
            return _onFocus.apply(this, arguments);
          }

          onFocus.toString = function () {
            return _onFocus.toString();
          };

          return onFocus;
        }(function (e) {
          if (onFocus) {
            onFocus(e);
          }
        }),
        "aria-label": labelText,
        id: id,
        required: required,
        disabled: disabled
      }, otherProps));
    }
  }]);
  return ControlledInput;
}(_react.Component);

exports.ControlledInput = ControlledInput;
ControlledInput.displayName = "ControlledInput";
(0, _defineProperty2.default)(ControlledInput, "propTypes", {
  extraClassNames: _propTypes.default.string,
  id: _propTypes.default.string,
  required: _propTypes.default.bool,
  labelText: _propTypes.default.string.isRequired,
  testId: _propTypes.default.string,
  checked: _propTypes.default.bool,
  onChange: _propTypes.default.func,
  name: _propTypes.default.string,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func,
  value: _propTypes.default.string,
  disabled: _propTypes.default.bool,
  type: _propTypes.default.oneOf(['checkbox', 'radio'])
});
(0, _defineProperty2.default)(ControlledInput, "defaultProps", {
  testId: 'cf-ui-controlled-input',
  required: false,
  disabled: false,
  type: 'checkbox'
});
var _default = ControlledInput;
exports.default = _default;
