// @contentful/forma-36-react-components/dist/components/Modal/Modal/Modal.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Modal = exports.ModalSizes = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactModal = _interopRequireDefault(require("react-modal"));

var _ModalHeader = _interopRequireDefault(require("../ModalHeader"));

var _ModalContent = _interopRequireDefault(require("../ModalContent"));

var _ModalControls = _interopRequireDefault(require("../ModalControls"));

var _ModalSizesMapper;

var styles = {
  "Modal__portal": "Modal__Modal__portal___F-liP",
  "Modal__overlay": "Modal__Modal__overlay___38PZk",
  "Modal__overlay--after-open": "Modal__Modal__overlay--after-open___2WVa7",
  "Modal__overlay--before-close": "Modal__Modal__overlay--before-close___1ZC0t",
  "Modal__overlay--centered": "Modal__Modal__overlay--centered___19mI8",
  "Modal__wrap": "Modal__Modal__wrap___16arr",
  "Modal__wrap--after-open": "Modal__Modal__wrap--after-open___2mXQz",
  "Modal": "Modal__Modal___1064L",
  "Modal__wrap--before-close": "Modal__Modal__wrap--before-close___143Hm",
  "Modal--overflow": "Modal__Modal--overflow___1aby9"
};
var ModalPositions = {
  CENTER: 'center',
  TOP: 'top'
};
var ModalSizes = {
  MEDIUM: 'medium',
  SMALL: 'small',
  LARGE: 'large'
};
exports.ModalSizes = ModalSizes;
var ModalSizesMapper = (_ModalSizesMapper = {}, (0, _defineProperty2.default)(_ModalSizesMapper, ModalSizes.MEDIUM, '520px'), (0, _defineProperty2.default)(_ModalSizesMapper, ModalSizes.SMALL, '400px'), (0, _defineProperty2.default)(_ModalSizesMapper, ModalSizes.LARGE, '700px'), _ModalSizesMapper);

var Modal =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Modal, _Component);

  function Modal() {
    (0, _classCallCheck2.default)(this, Modal);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Modal).apply(this, arguments));
  }

  (0, _createClass2.default)(Modal, [{
    key: "renderDefault",
    value: function renderDefault() {
      return _react.default.createElement(_react.default.Fragment, null, this.props.title && _react.default.createElement(_ModalHeader.default, {
        title: this.props.title,
        onClose: this.props.onClose,
        isNotWrapped: this.props.isNotWrapped
      }), _react.default.createElement(_ModalContent.default, null, this.props.children));
    }
  }, {
    key: "render",
    value: function render() {
      var _cn;

      return _react.default.createElement(_reactModal.default, {
        ariaHideApp: false,
        onRequestClose: this.props.onClose,
        isOpen: this.props.isShown,
        onAfterOpen: this.props.onAfterOpen,
        shouldCloseOnEsc: this.props.shouldCloseOnEscapePress,
        shouldCloseOnOverlayClick: this.props.shouldCloseOnOverlayClick,
        portalClassName: styles.Modal__portal,
        className: {
          base: styles.Modal__wrap,
          afterOpen: styles['Modal__wrap--after-open'],
          beforeClose: styles['Modal__wrap--before-close']
        },
        style: {
          content: {
            top: this.props.position === ModalPositions.CENTER ? 0 : this.props.topOffset
          }
        },
        overlayClassName: {
          base: (0, _classnames.default)((_cn = {}, (0, _defineProperty2.default)(_cn, styles.Modal__overlay, true), (0, _defineProperty2.default)(_cn, styles['Modal__overlay--centered'], this.props.position === ModalPositions.CENTER), _cn)),
          afterOpen: styles['Modal__overlay--after-open'],
          beforeClose: styles['Modal__overlay--before-close']
        },
        htmlOpenClassName: "Modal__html--open",
        bodyOpenClassName: "Modal__body--open",
        closeTimeoutMS: 300
      }, _react.default.createElement("div", {
        "data-test-id": this.props.testId,
        style: {
          width: ModalSizesMapper[this.props.size] || this.props.size
        },
        className: (0, _classnames.default)(styles.Modal, this.props.extraClassNames, (0, _defineProperty2.default)({}, styles['Modal--overflow'], this.props.allowHeightOverflow))
      }, typeof this.props.children === 'function' ? this.props.children(this.props) : this.renderDefault()));
    }
  }]);
  return Modal;
}(_react.Component);

exports.Modal = Modal;
Modal.displayName = "Modal";
(0, _defineProperty2.default)(Modal, "propTypes", {
  isShown: _propTypes.default.bool.isRequired,
  onClose: _propTypes.default.any.isRequired,
  onAfterOpen: _propTypes.default.any,
  shouldCloseOnOverlayClick: _propTypes.default.bool,
  shouldCloseOnEscapePress: _propTypes.default.bool,
  position: _propTypes.default.oneOf(['center', 'top']),
  topOffset: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]),
  title: _propTypes.default.string,
  size: _propTypes.default.oneOfType([_propTypes.default.oneOf(['small']), _propTypes.default.oneOf(['medium']), _propTypes.default.oneOf(['large']), _propTypes.default.string, _propTypes.default.number]),
  allowHeightOverflow: _propTypes.default.bool,
  isNotWrapped: _propTypes.default.bool,
  extraClassNames: _propTypes.default.string,
  testId: _propTypes.default.string,
  children: _propTypes.default.any.isRequired
});
(0, _defineProperty2.default)(Modal, "Positions", ModalPositions);
(0, _defineProperty2.default)(Modal, "Sizes", ModalSizes);
(0, _defineProperty2.default)(Modal, "Header", _ModalHeader.default);
(0, _defineProperty2.default)(Modal, "Content", _ModalContent.default);
(0, _defineProperty2.default)(Modal, "Controls", _ModalControls.default);
(0, _defineProperty2.default)(Modal, "defaultProps", {
  shouldCloseOnEscapePress: true,
  shouldCloseOnOverlayClick: true,
  position: 'center',
  testId: 'cf-ui-modal',
  topOffset: '50px',
  size: 'medium',
  allowHeightOverflow: false,
  onAfterOpen: function onAfterOpen() {}
});
var _default = Modal;
exports.default = _default;
