// @contentful/forma-36-react-components/dist/components/Note/Note.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Note = exports.NoteType = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = _interopRequireDefault(require("../Icon/Icon"));

var _constants = require("../Icon/constants");

var _Icons;

var styles = {
  "Note": "Note__Note___2eSKN",
  "Note--primary": "Note__Note--primary___-8-7D",
  "Note--negative": "Note__Note--negative___3fkwL",
  "Note--positive": "Note__Note--positive___3Omoh",
  "Note--warning": "Note__Note--warning___3X53I",
  "Note__title": "Note__Note__title___2nwpx",
  "Note__icon": "Note__Note__icon___20RqC"
}; // had to use 'as' here in a weird way, due to backward compatible reasons
// since NoteType is exposted as an external API and keys names cannot be changed with a breaking change

var NoteType = {
  PRIMARY: 'primary',
  POSITIVE: 'positive',
  NEGATIVE: 'negative',
  WARNING: 'warning'
};
exports.NoteType = NoteType;
var Icons = (_Icons = {}, (0, _defineProperty2.default)(_Icons, NoteType.PRIMARY, _constants.iconName.InfoCircle), (0, _defineProperty2.default)(_Icons, NoteType.POSITIVE, _constants.iconName.CheckCircle), (0, _defineProperty2.default)(_Icons, NoteType.NEGATIVE, _constants.iconName.Warning), (0, _defineProperty2.default)(_Icons, NoteType.WARNING, _constants.iconName.Warning), _Icons);

var Note =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Note, _Component);

  function Note() {
    (0, _classCallCheck2.default)(this, Note);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Note).apply(this, arguments));
  }

  (0, _createClass2.default)(Note, [{
    key: "render",
    value: function render() {
      var _classNames;

      var icon = Icons[this.props.noteType];

      if (!icon) {
        throw new Error("Intent ".concat(this.props.noteType, " is not supported in Note component."));
      }

      return _react.default.createElement("div", {
        style: this.props.style,
        className: (0, _classnames.default)(styles.Note, this.props.extraClassNames, (_classNames = {}, (0, _defineProperty2.default)(_classNames, styles['Note--primary'], this.props.noteType === NoteType.PRIMARY), (0, _defineProperty2.default)(_classNames, styles['Note--positive'], this.props.noteType === NoteType.POSITIVE), (0, _defineProperty2.default)(_classNames, styles['Note--negative'], this.props.noteType === NoteType.NEGATIVE), (0, _defineProperty2.default)(_classNames, styles['Note--warning'], this.props.noteType === NoteType.WARNING), _classNames)),
        "data-test-id": this.props.testId
      }, _react.default.createElement("div", {
        className: styles.Note__icon
      }, _react.default.createElement(_Icon.default, {
        icon: icon,
        color: this.props.noteType
      })), _react.default.createElement("div", {
        className: styles.Note__info
      }, this.props.title && _react.default.createElement("div", {
        className: styles.Note__title
      }, this.props.title), _react.default.createElement("div", null, this.props.children)));
    }
  }]);
  return Note;
}(_react.Component);

exports.Note = Note;
Note.displayName = "Note";
(0, _defineProperty2.default)(Note, "propTypes", {
  noteType: _propTypes.default.oneOf(['primary', 'positive', 'negative', 'warning']),
  extraClassNames: _propTypes.default.string,
  title: _propTypes.default.string,
  style: _propTypes.default.any,
  testId: _propTypes.default.string,
  children: _propTypes.default.node.isRequired
});
(0, _defineProperty2.default)(Note, "Type", NoteType);
(0, _defineProperty2.default)(Note, "defaultProps", {
  noteType: NoteType.PRIMARY,
  testId: 'cf-ui-note'
});
var _default = Note;
exports.default = _default;
