// contentful-migration/built/lib/intent-validator/entry-transform-to-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const schema_validator_1 = require("./schema-validator");
const Joi = require("joi");
class EntryTransformToTypeIntentValidator extends schema_validator_1.default {
    constructor() {
        super(...arguments);
        this.article = 'an';
        this.displayName = 'entry derivation';
    }
    appliesTo(step) {
        return step.isEntryDerive();
    }
    // NOTE: this could be change to return the object to
    // validate against the schema instead of returning
    // just the prop name
    get propertyNameToValidate() {
        return 'derivation';
    }
    get schema() {
        return {
            sourceContentType: Joi.string().required(),
            targetContentType: Joi.string().required(),
            from: Joi.array().items(Joi.string()).required(),
            to: Joi.array().items(Joi.string()).required(),
            identityKey: Joi.func().required(),
            shouldPublish: Joi.boolean(),
            deriveEntryForLocale: Joi.func().required()
        };
    }
}
exports.default = EntryTransformToTypeIntentValidator;
